tools/src/main/scala/scala/scalanative/linker/Linker.scala

```diff
@@ -89,18 +89,12 @@ object Linker {
       (unresolved.toSeq, links.toSeq, defns.sortBy(_.name.toString).toSeq)
     }
 
-    def link(entries: Seq[Global]): (Seq[Attr.Link], Seq[Defn]) = {
+    def link(entries: Seq[Global]): (Seq[Global], Seq[Attr.Link], Seq[Defn]) = {
       val (unresolved, links, defns) = impl(entries)
 
       config.paths.foreach(_.close)
```

Comment:
Those should go to `def close` on the linker itself. Then `def impl` can just be renamed to `def link`.